pkgs/development/python-modules/pygreat/default.nix

```diff
@@ -1,41 +1,52 @@
 {
   lib,
   buildPythonPackage,
-  isPy3k,
-  fetchFromGitHub,
+  fetchPypi,
+  setuptools,
+  pythonOlder,
+  pythonRelaxDepsHook,
   future,
   pyusb,
 }:
 
-buildPythonPackage {
+buildPythonPackage rec {
   pname = "pygreat";
-  version = "2019.5.1.dev0";
-  format = "setuptools";
-
-  src = fetchFromGitHub {
-    owner = "greatscottgadgets";
-    repo = "libgreat";
-    rev = "14c00b7c8f036f4d467e4b1a324ffa3566b126fa";
-    sha256 = "1h0z83k1k4z8j36z936h61l8j3cjr3wsxr86k91v5c5h93g9dkqh";
+  version = "2024.0.2";
+  pyproject = true;
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-VGZyvGLPyA6tOVh7boz4j/6o/iQ9mRvULOleFypGHGY=";
   };
 
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace-fail '"setuptools-git-versioning<2"' "" \
+      --replace-fail 'dynamic = ["version"]' 'version = "${version}"'
+  '';
+
+  build-system = [
+    setuptools
+  ];
+
+  nativeBuildInputs = [
+    pythonRelaxDepsHook
+  ];
+
+  pythonRemoveDeps = [
+    "backports.functools-lru-cache"
+  ];
+
   propagatedBuildInputs = [
     future
     pyusb
   ];
 
```

Comment:
Could you enable tests with `pytestCheckHook` and `pythonImportsCheck`?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

Reply:
I added `pythonImportsCheck`, but this package doesn't have any tests for `pytestCheckHook` currently.

Reply:
Can we set doCheck explicit to false?

Reply:
Done!